contracts/RenPool.sol

```diff
@@ -356,5 +344,21 @@ contract RenPool {
                     );
 										See: https://github.com/renproject/gateway-sol/blob/7bd51d8a897952a31134875d7b2b621e4542deaa/test/Gateway.ts
 		*/
+
+    /*
+        // Construct the payload hash and mint new tokens using the Gateway
+        // contract. This will verify the signature to ensure the Darknodes have
+        // received the Bitcoin.
+        bytes32 pHash =
+            keccak256(abi.encode(_beneficiary, _startTime, _duration));
+        uint256 finalAmountScaled =
+            registry.getGatewayBySymbol("BTC").mint(
+                pHash,
+                _amount,
+                _nHash,
+                _sig
```

Comment:
Not sure if this helps, but `sigString` variable as defined here[[1]](https://github.com/renproject/gateway-sol/blob/7bd51d8a897952a31134875d7b2b621e4542deaa/test/Vesting.ts#L101-L115)

```js
                const hashForSignature = await btcGateway.hashForSignature.call(
                    Ox(pHash),
                    amount.toNumber(),
                    vesting.address,
                    nHash
                );
                const sig = ecsign(
                    Buffer.from(hashForSignature.slice(2), "hex"),
                    privKey
                );
                const sigString = Ox(
                    `${sig.r.toString("hex")}${sig.s.toString(
                        "hex"
                    )}${sig.v.toString(16)}`
                );
```

flows into the `mint` method[[2]](https://github.com/renproject/gateway-sol/blob/master/contracts/Gateway/examples/Vesting.sol#L73-L78)

```js
            registry.getGatewayBySymbol("BTC").mint(
                pHash,
                _amount,
                _nHash,
                _sig
            );
```

Reply:
As discussed, this seems to be the approach we need to follow but, since we are registering the darknode via a contract, we don't have a notion of private keys and therefore we need to wait for the ren team to give us some alternative apporach to sign the request